orchestrator-salt/src/main/resources/salt/salt/nginx/conf/clouderamanager-ssl-user-facing.conf

```diff
@@ -28,6 +28,14 @@ server {
         rewrite ^([/](?!clouderamanager/).*$) /clouderamanager$1;
     }
 
+    # If hadoop-jwt cookie not set and favicon is requested return 403.
```

Comment:
So let me see if I understand what this change is doing. Before this change, the browser made two requests to knox that is fronted by this nginx, one for the actual resource and right afterwards for the favicon.ico. When google changed the default SameSite default the pac4j cookie was not sent by the browser so knox did not know where to redirect the browser to. And then when the second request came it just returned the favicon and that is what you got. 

What this change is doing is returning a 403 for the second request. This means that the user will see a 403 UNAUTHORIZED error message on the screen instead of the favicon, right?

Reply:
Not exactly, browser will request (on its own) favicon.ico for new websites. Now, what happens is when we click the UI link from Control Plane we get redirected to the UI page with a request for the resource (CM, Ranger, Atlas etc.), Chrome will, on its own also request favicon.ico. These two requests would go through full KnoxSSO cycle and for some unknown reason Altus would return back response for favicon.ico (even when favicon.ico was the send request) and drop the request for requested resource. We saw this in the SAML debugging session we had few weeks ago. How this is related to SameSite is honestly a mystery to me :(  

What the fix tries to do is block favicon.ico requests for unauthenticated requests. 
Now since favicon.ico request is initiated by the browser users will not see 403. And upon successful authentication browser will again request favicon.ico and this time it will succeed. This is demonstrated in the screenshot I pasted above.

Reply:
The control plane gets exactly one SAML authentication request from Knox and it's never for favicon. I believe that after you set the samesite property in knox and atlas we could not reproduce the issue any more. That said, @risdenk thinks that there are no issues with SameSite as he is running with the new feature enabled and haven't encountered any problem. 

I would like to get to the bottom of this if possible. Let's chat offline.

Reply:
After looking at a reproduction with you again, I agree that not triggering the authentication route for favicon requests is the right thing to do. I am not sure that returning a 403 is the right thing as it seems like the browser will never retry the request. But maybe I'm wrong and maybe it's not important. I added @arnebech and @joeheyming as reviewers as they may have more ideas/information re: what to return to the original favicon request.